src/Microsoft.ComponentDetection.Orchestrator/Services/ScanExecutionService.cs

```diff
@@ -38,14 +38,17 @@ public async Task<ScanResult> ExecuteScanAsync(ScanSettings settings)
         using var scope = this.logger.BeginScope("Executing BCDE scan");
 
         var detectorRestrictions = this.GetDetectorRestrictions(settings);
-        var detectors = this.detectorRestrictionService.ApplyRestrictions(detectorRestrictions, this.detectors).ToImmutableList();
+        var detectors = this.detectors.ToList();
+        var restrictedDetectors = this.detectorRestrictionService.ApplyRestrictions(detectorRestrictions, detectors).ToImmutableList();
+        detectorRestrictions.DisabledDetectors = detectors.Except(restrictedDetectors).ToList();
 
         this.logger.LogDebug("Finished applying restrictions to detectors.");
 
-        var processingResult = await this.detectorProcessingService.ProcessDetectorsAsync(settings, detectors, detectorRestrictions);
+        var processingResult = await this.detectorProcessingService.ProcessDetectorsAsync(settings, restrictedDetectors, detectorRestrictions);
```

Comment:
nit: This naming seems a bit confusing. We need to process all the detectors. This is just a subset of detectors, but naming it like this + the `detectorRestrictions` next to it makes it confusing as to what are we actually passing.